### PR TITLE
Update RuleEntitlement to save data as bytes

### PR DESCRIPTION
**Pull request by giuseppecrj**

### CODE REVIEW

File: contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol [hunk 1]
```diff
@@ -190,6 +191,8 @@ contract RuleEntitlementV2 is
     }
 
     delete entitlementsByRoleId[roleId];
+    delete entitlementsByRoleId[roleId].grantedBy;
```
Comment (brianathere) [inline contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol:194]: These need to be before the 

```
delete entitlementsByRoleId[roleId];
```

**Review by brianathere — APPROVED** [review]
Comment inline.

Why do we want to change this?

**Comment by giuseppecrj** [comment]
> Comment inline.
> 
> Why do we want to change this?

If we're in accordance into how we save data, I want to move to using a namespaced storage approach as per EIP-7201
https://eips.ethereum.org/EIPS/eip-7201

**Comment by brianathere** [comment]
> > Comment inline.
> > Why do we want to change this?
> 
> If we're in accordance in how we save data, I want to move to using a namespaced storage approach as per EIP-7201 https://eips.ethereum.org/EIPS/eip-7201

Thanks, I'll read. I assume it makes sense :-)

File: contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol [hunk 2]
```diff
@@ -38,17 +37,20 @@ contract RuleEntitlementV2 is
...
     address grantedBy;
     uint256 grantedTime;
-    RuleData data;
+    bytes data;
   }
 
-  mapping(uint256 => Entitlement) internal entitlementsByRoleId;
-
-  address public SPACE_ADDRESS;
+  struct Layout {
```
Comment (bas-vk) [inline contracts/src/spaces/entitlements/rule/RuleEntitlementV2.sol:50]: Why not annotate `Layout` with `@custom:storage-location` and bump the solc version to ^0.8.20 as erc7201 describes?

**Pull request closed without merging**